models/phoenix/phoenix_v1_trades.sql

```diff
@@ -0,0 +1,236 @@
+        SELECT 
+            tkA.symbol as tokenA_symbol
+            , length(json_extract_scalar(initializeParams, '$.InitializeParams.numBaseLotsPerBaseUnit')) - 1 as tokenA_decimals --if lot size is 1000, then its 3 decimals
+            , ip.account_baseMint as tokenA
+            , ip.account_baseVault as tokenAVault
+            , tkB.symbol as tokenB_symbol
+            , length(json_extract_scalar(initializeParams, '$.InitializeParams.numQuoteLotsPerQuoteUnit')) - 1 as tokenB_decimals
+            , ip.account_quoteMint as tokenB
+            , ip.account_quoteVault as tokenBVault
+            , cast(json_extract_scalar(initializeParams, '$.InitializeParams.takerFeeBps') as double)/100 as fee_tier
+            , ip.account_market as pool_id
+            , ip.call_tx_id as init_tx
+        FROM {{ source('phoenix_v1_solana','phoenix_v1_call_InitializeMarket') }} ip
+        LEFT JOIN {{ ref('tokens_solana_fungible') }}  tkA ON tkA.token_mint_address = ip.account_baseMint
+        LEFT JOIN {{ ref('tokens_solana_fungible') }}  tkB ON tkB.token_mint_address = ip.account_quoteMint
+  )
+  
+  , logs AS (
+    SELECT
+      call_tx_id,
+      call_is_inner,
+      call_block_slot,
+      call_block_time,
+      call_inner_instruction_index,
+      call_outer_instruction_index,
+      call_tx_index,
+      BYTEARRAY_TO_BIGINT (
+        BYTEARRAY_REVERSE (BYTEARRAY_SUBSTRING (l.call_data, 4, 8))
+      ) AS seq,
+      TO_BASE58 ((BYTEARRAY_SUBSTRING (l.call_data, 28, 32))) AS market,
+      cast(BYTEARRAY_TO_BIGINT (
+        BYTEARRAY_REVERSE (
+          BYTEARRAY_SUBSTRING (
+            l.call_data,
+            BYTEARRAY_LENGTH (l.call_data) - 23,
+            8
+          )
+        )
+      ) as uint256) AS tokenA_filled,
+      l.call_outer_instruction_index AS index,
+      cast(BYTEARRAY_TO_BIGINT (
+        BYTEARRAY_REVERSE (
+          BYTEARRAY_SUBSTRING (
+            l.call_data,
+            BYTEARRAY_LENGTH (l.call_data) - 15,
+            8
+          )
+        )
+      ) as uint256) AS tokenB_filled
+    FROM
+      {{ source('phoenix_v1_solana','phoenix_v1_call_Log')}} AS l
+    WHERE 1=1
+      --filter for 0 events
+      and bytearray_length (l.call_data) > 93
+      --instruction is swap
+      and BYTEARRAY_TO_BIGINT (
+        BYTEARRAY_REVERSE (BYTEARRAY_SUBSTRING (l.call_data, 3, 1))
+      ) = 0
+      --filter for trade size > 0
+      AND BYTEARRAY_TO_BIGINT (
+        BYTEARRAY_REVERSE (
+          BYTEARRAY_SUBSTRING (
+            l.call_data,
+            BYTEARRAY_LENGTH (l.call_data) - 23,
+            8
+          )
+        )
+      ) > 0
+      --filter for event FillSummary
+      AND BYTEARRAY_TO_BIGINT (
+        BYTEARRAY_REVERSE (
+          BYTEARRAY_SUBSTRING (
+            l.call_data,
+            BYTEARRAY_LENGTH (l.call_data) - 42,
+            1
+          )
+        )
+      ) = 6
+      {% if is_incremental() %}
+      AND {{incremental_predicate('call_block_time')}}
+      {% endif %}
```

Comment:
@aalan3 this one

Reply:
try using `l. call_block_time` here? but this should be working.

Reply:
the line numbers you see in the failure doesn't map very well to the uncompiled sql

Reply:
okay, trying CI again

Reply:
alright it worked this time!